prime-router/src/main/kotlin/fhirengine/engine/QueueMessage.kt

```diff
@@ -85,33 +91,67 @@ data class RawSubmission(
     override val blobSubFolderName: String,
     override val topic: Topic,
     val schemaName: String = "",
-) : Message()
+) : UniversalPipelineQueueMessage()
 
 @JsonTypeName("convert")
-data class FhirConvertMessage(
+data class FhirConvertQueueMessage(
     override val reportId: ReportId,
     override val blobURL: String,
     override val digest: String,
     override val blobSubFolderName: String,
     override val topic: Topic,
     val schemaName: String = "",
-) : Message()
+) : UniversalPipelineQueueMessage()
 
 @JsonTypeName("route")
-data class FhirRouteMessage(
+data class FhirRouteQueueMessage(
     override val reportId: ReportId,
     override val blobURL: String,
     override val digest: String,
     override val blobSubFolderName: String,
     override val topic: Topic,
-) : Message()
+) : UniversalPipelineQueueMessage()
 
 @JsonTypeName("translate")
-data class FhirTranslateMessage(
+data class FhirTranslateQueueMessage(
     override val reportId: ReportId,
     override val blobURL: String,
     override val digest: String,
     override val blobSubFolderName: String,
     override val topic: Topic,
     val receiverFullName: String,
-) : Message()
\ No newline at end of file
+) : UniversalPipelineQueueMessage()
+
+abstract class MixedPipelineQueueMessage : QueueMessage() {
+    abstract val eventType: String
+    abstract val eventAction: Event.EventAction
+}
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+data class BatchEventQueueMessage(
+    @JsonProperty("eventType") override val eventType: String,
+    @JsonProperty("eventAction") override val eventAction: Event.EventAction,
```

Comment:
I think you can do away with property now since the need for it gets replaced by the fact that we've captured it in the type.  If the code does need it in the short term, I think you can just add a method like `getEventAction` to the abstract class.

I think the same might go for `eventType`

Reply:
EventType yes, EventAction no.